cdap-common/src/main/java/co/cask/cdap/common/utils/TimeMathParser.java

```diff
@@ -79,20 +82,57 @@ public static int resolutionInSeconds(String resolutionStr) {
     return output;
   }
 
+  /**
+   * Parses a time in String format into a long value. If the input is numeric we assume the input is in seconds.
+   *
+   * @param timeStr the string to parse
+   * @return the parsed time in seconds
+   */
   public static long parseTime(String timeStr) {
-    return parseTime(nowInSeconds(), timeStr);
+    return parseTime(timeStr, TimeUnit.SECONDS);
   }
 
+  /**
+   * Parses a time in String format into a long value
+   *
+   * @param timeStr the string to parse
+   * @param timeUnit the unit of time to return, if timeStr is numeric then it is assumed to be in the unit timeUnit
+   * @return the parsed time
+   */
+  public static long parseTime(String timeStr, TimeUnit timeUnit) {
+    return parseTime(nowInSeconds(), timeStr, timeUnit);
+  }
+
+  /**
+   * Parses a time in String format into a long value. If the input is numeric we assume the input is in seconds.
+   *
+   * @param now the present time in seconds
+   * @param timeStr the string to parse
+   * @return the parsed time in seconds
+   */
   public static long parseTime(long now, String timeStr) {
+    return parseTime(now, timeStr, TimeUnit.SECONDS);
+  }
+
+  /**
+   * Parses a time in String format into a long value
+   *
+   * @param now the present time in seconds
+   * @param timeStr the string to parse
+   * @param timeUnit the unit of time to return, if timeStr is numeric then it is assumed to be in the unit timeUnit
+   * @return the parsed time
+   * @throws IllegalArgumentException if the format of timeStr is bad
+   */
+  public static long parseTime(long now, String timeStr, TimeUnit timeUnit) {
```

Comment:
you are adding this method so you can get the time in milliseconds and other time units right?. in the usage i see its mostly used with mill seconds as the timeUnit, but the timeStr string's valid units are only {s | m | h | d}, would it help to allow the parser to include milliseconds ?  - the time math parser was used only with metrics before and that's why seconds is the lowest granularity, for stream we might want to include milli seconds. what do you think?